### DISCUSSION TITLE
Specify a unique number of output formats

### DISCUSSION THREAD

**Discussion by MattPenaTV** [openai/whisper#1869] [Q&A]
From "whisper --help":

```
  --output_format {txt,vtt,srt,tsv,json,all}, -f {txt,vtt,srt,tsv,json,all}
                        format of the output file; if not specified, all available formats will be produced (default:
                        all)
```

It's clear that I can specify to use one output format (ex srt), or all output formats (all), but what if I want to use only 2 output formats (ie srt,json)? Is that possible to specify?

Cheers

**Comment by ryanheise** [comment]
Unfortunately not at this time, although there is a PR addressing this: #1039

**Reply by MattPenaTV** [reply]
What do you think the best work-around would be, for speed and quality?
1) use {all}, and delete everything except {src} and {json}
2) run twice, once with {src} and once with {json}
3) something else

**Reply by glangford** [reply]
> What do you think the best work-around would be, for speed and quality?

Option 1 with a model size of your choosing.

**Reply by MattPenaTV** [reply]
> Option 1 with a model size of your choosing.

Thank you. Why did you bring up the model size? Is that relevant here?

**Reply by glangford** [reply]
You said "for speed and quality" - a large model will run slower but give you better results.

**Reply by ryanheise** [reply]
It's something like this: 99.999% of the processing time is spent running the model to do the actual transcription, and 0.001% of the processing time to convert that to different output formats. So if you run twice, you're running the model twice which is going to really cost you a lot of time. But if you run once with "all" outputs, the cost will be negligible because generating the outputs takes almost no time anyway.
